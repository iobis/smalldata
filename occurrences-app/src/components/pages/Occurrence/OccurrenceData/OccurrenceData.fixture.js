import OccurrenceData from './OccurrenceData'

export default [{
  component:   OccurrenceData,
  name:        'no selection step',
  props:       {
    data:     {
      basisOfRecord:    'humanObservation',
      beginDate:        Date.now(),
      endDate:          null,
      lifestage:        null,
      occurrenceStatus: 'present',
      scientificName:   '',
      sex:              null
    }
  },
  controllers: {
    onChange: data =>({ data })
  }
}, {
  component: OccurrenceData,
  name:      'with selections',
  props:     {
    onChange: console.log,
    data:     {
      basisOfRecord:    'humanObservation',
      beginDate:        new Date(),
      endDate:          new Date(),
      lifestage:        'adult',
      occurrenceStatus: 'absent',
      scientificName:   'ala abra',
      sex:              'male'
    }
  }
}]
