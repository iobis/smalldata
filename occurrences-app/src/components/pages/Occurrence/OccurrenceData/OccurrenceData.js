import CopyPreviousData from '../CopyPreviousData'
import DatePicker from '../../../form/DatePicker'
import InputRadioGroup from '../../../form/InputRadioGroup'
import PropTypes from 'prop-types'
import React, { useEffect, useState, useRef } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as MarineSpeciesClient from '../../../../clients/MarineSpeciesClient'
import { useDebounce, useOnClickOutside } from '../../../../hooks/hooks'
import { useTranslation } from 'react-i18next'
import classNames from 'classnames'

const basisOfRecordOptions = ['humanObservation', 'fossilSpecimen', 'livingSpecimen', 'machineSpecimen', 'preservedSpecimen']
const lifestageOptions = ['egg', 'eft', 'juvenile', 'adult', 'unspecified']
const occurrenceStatusOptions = ['absent', 'present']
const sexOptions = ['male', 'female', 'hermaphrodite', 'unspecified']

export default function OccurrenceData({ onChange, data }) {
  const { t } = useTranslation()
  const { basisOfRecord, beginDate, endDate, lifestage, occurrenceStatus, scientificName, sex } = data

  const updateField = (name, value) => {
    const newSelection = { ...data, [name]: value }
    onChange(newSelection)
  }

  return (
    <div className="occurrence-data section is-fluid">
      <div className="columns">
        <ScientificNameInput
          onChange={(value) => updateField('scientificName', value)}
          scientificName={scientificName}/>
      </div>
      <div className="columns">
        <div className="event-begin-date column field is-two-fifths">
          <label className="label">
            {t('occurrenceForm.occurrenceData.eventBeginDate')}
          </label>
          <DatePicker onChange={(value) => updateField('beginDate', value)} value={beginDate}/>
        </div>
        <div className="event-end-date column field is-two-fifths">
          <label className="label">
            {t('occurrenceForm.occurrenceData.eventEndDate')}
          </label>
          <div className="control">
            <DatePicker onChange={(value) => updateField('endDate', value)} value={endDate}/>
          </div>
          <p className="help">{t('occurrenceForm.occurrenceData.eventEndDateHelp')}</p>
        </div>
      </div>
      <InputRadioGroup
        name="occurrenceForm.occurrenceData.occurrenceStatus"
        onChange={(value) => updateField('occurrenceStatus', value)}
        options={occurrenceStatusOptions}
        selectedValue={occurrenceStatus}/>
      <InputRadioGroup
        name="occurrenceForm.occurrenceData.basisOfRecord"
        onChange={(value) => updateField('basisOfRecord', value)}
        options={basisOfRecordOptions}
        selectedValue={basisOfRecord}/>
      <InputRadioGroup
        name="occurrenceForm.occurrenceData.sex"
        onChange={(value) => updateField('sex', value)}
        options={sexOptions}
        selectedValue={sex}/>
      <InputRadioGroup
        name="occurrenceForm.occurrenceData.lifestage"
        onChange={(value) => updateField('lifestage', value)}
        options={lifestageOptions}
        selectedValue={lifestage}/>
      <CopyPreviousData/>
    </div>
  )
}

OccurrenceData.propTypes = {
  data:     PropTypes.shape({
    basisOfRecord:    PropTypes.oneOf(basisOfRecordOptions).isRequired,
    beginDate:        PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.number]).isRequired,
    endDate:          PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.number]),
    lifestage:        PropTypes.oneOf(lifestageOptions),
    occurrenceStatus: PropTypes.oneOf(occurrenceStatusOptions).isRequired,
    scientificName:   PropTypes.string.isRequired,
    sex:              PropTypes.oneOf(sexOptions)
  }).isRequired,
  onChange: PropTypes.func.isRequired
}

function ScientificNameInput({ scientificName, onChange }) {
  const { t } = useTranslation()
  const ref = useRef()
  const [firstRender, setFirstRender] = useState(true)
  const [loading, setLoading] = useState(false)
  const [name, setName] = useState(scientificName)
  const [nameValid, setNameValid] = useState(false)
  const [suggestions, setSuggestions] = useState([])
  const [dropdownActive, setDropdownActive] = useState(false)
  const hideDropdownOptions = () => setDropdownActive(false)
  const showDropdownOption = () => {if (dropdownActive === false) setDropdownActive(true)}
  const debouncedName = useDebounce(name, 500)
  const isRecordWithName = (record, name) => (record.scientificname || '').trim().toLowerCase() === (name || '').trim().toLowerCase()
  const findRecordWithName = (records, name) => records.find(record => isRecordWithName(record, name))

  useOnClickOutside(ref, hideDropdownOptions)

  useEffect(() => {
    const getByName = async() => {
      setLoading(true)
      const newSuggestions = await MarineSpeciesClient.getByName(name)
      const newNameValid = !!findRecordWithName(newSuggestions, name)
      setSuggestions(newSuggestions)
      setNameValid(newNameValid)
      if (!newNameValid && newSuggestions.length > 0) showDropdownOption()
      setLoading(false)
      setFirstRender(false)
    }

    if (debouncedName) getByName()
  }, [debouncedName])

  function handleNameChange(newName) {
    onChange(newName)
    setName(newName)
  }

  function handleSuggestionClick(newName) {
    hideDropdownOptions()
    onChange(newName)
    setName(newName)
  }

  return (
    <div className="field column is-four-fifths" ref={ref}>
      <div className={classNames('dropdown', { 'is-active': dropdownActive && suggestions.length > 0 })}>
        <div className="dropdown-trigger">
          <label className="label">{t('occurrenceForm.occurrenceData.scientificName')}</label>
          <div className={classNames('control has-icons-right', { 'is-loading': loading })}>
            <input
              className={classNames('input', { 'is-danger': !firstRender && !nameValid && suggestions.length === 0 })}
              onChange={(value) => handleNameChange(value.target.value)}
              onClick={showDropdownOption}
              placeholder={t('occurrenceForm.occurrenceData.scientificName')}
              type="text"
              value={name}/>
            {nameValid && !loading
              ? <span className="clear icon is-small is-right"><FontAwesomeIcon className="check" icon="check"/></span>
              : null}
          </div>
        </div>
        <div className="dropdown-menu" id="dropdown-menu" role="menu">
          <div className="dropdown-content">
            {suggestions.map(record => (
              <div
                className={classNames('dropdown-item', { 'is-active': isRecordWithName(record, name) })}
                key={record.lsid}
                onClick={() => handleSuggestionClick(record.scientificname)}>
                {record.scientificname}
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}

ScientificNameInput.propTypes = {
  onChange:       PropTypes.func.isRequired,
  scientificName: PropTypes.string.isRequired
}
