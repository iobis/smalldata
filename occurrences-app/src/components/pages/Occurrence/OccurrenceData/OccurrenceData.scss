.occurrence-data {
  margin-bottom: 1.25rem;

  .dropdown {
    display: block;

    .dropdown-content {
      max-height: 15em;
      overflow: auto;
    }

    .dropdown-menu {
      width: 100%;
    }
  }

  .field {
    .label {
      small {
        font-size: 0.75em;
        font-style: italic;
      }
    }
  }

  .event-begin-date {
    .datetimepicker-clear-button {
      display: none;
    }
  }

  .input-radio-group {
    margin-right: 180px;
  }
}

.worms-info {
  font-size: $size-7;
}

.scientific-name {
  position: relative;
  z-index: 1;
}

.scientific-name.is-danger {
  background-color: #ffeb3b;
  border-color: black;
}

.scientific-name.is-danger ~ .invalid-worms::after {
  content: 'invalid worms value, but will be submitted';
  position: absolute;
  z-index: 1000;
  padding: 0.2rem;
  margin-left: 0.4rem;
  font-size: $size-7;
  word-break: keep-all;
  white-space: nowrap;
  border-bottom: 3px solid orange;
  background-color: white;
}
