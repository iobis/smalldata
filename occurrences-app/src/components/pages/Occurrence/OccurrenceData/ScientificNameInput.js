import * as MarineSpeciesClient from '../../../../clients/MarineSpeciesClient'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import React, { useEffect, useRef, useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { useDebounce, useOnClickOutside } from '@smalldata/dwca-lib'
import { useTranslation } from 'react-i18next'

export default function ScientificNameInput({ scientificName, onChange }) {
  const { t } = useTranslation()
  const ref = useRef()
  const [firstRender, setFirstRender] = useState(true)
  const [loading, setLoading] = useState(false)
  const [name, setName] = useState(scientificName)
  const [nameValid, setNameValid] = useState(false)
  const [suggestions, setSuggestions] = useState([])
  const [dropdownActive, setDropdownActive] = useState(false)
  const hideDropdownOptions = () => setDropdownActive(false)
  const showDropdownOption = () => {if (dropdownActive === false) setDropdownActive(true)}
  const debouncedName = useDebounce(name, 500)
  const isRecordWithName = (record, name) => (record.scientificname || '').trim().toLowerCase() === (name || '').trim().toLowerCase()
  const findRecordWithName = (records, name) => records.find(record => isRecordWithName(record, name))

  useOnClickOutside(ref, hideDropdownOptions)

  useEffect(() => setName(scientificName), [scientificName])

  useEffect(() => {
    const getByName = async() => {
      setLoading(true)
      const newSuggestions = await MarineSpeciesClient.getByName(name)
      const newNameValid = !!findRecordWithName(newSuggestions, name)
      setSuggestions(newSuggestions)
      setNameValid(newNameValid)
      if (!newNameValid && newSuggestions.length > 0) showDropdownOption()
      setLoading(false)
      setFirstRender(false)
    }

    if (debouncedName) getByName()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [debouncedName])

  function handleNameChange(newName) {
    if (newName === '') setNameValid(false)
    onChange(newName)
    setName(newName)
  }

  function handleSuggestionClick(newName) {
    hideDropdownOptions()
    onChange(newName)
    setName(newName)
  }

  return (
    <div className="field column is-four-fifths" ref={ref}>
      <div className={classNames('dropdown', { 'is-active': dropdownActive && suggestions.length > 0 })}>
        <div className="dropdown-trigger">
          <label className="label">{t('occurrenceForm.occurrenceData.scientificName')}</label>
          <div className={classNames('control has-icons-right', { 'is-loading': loading })}>
            <input
              className={classNames('input scientific-name', { 'is-danger': !firstRender && !nameValid && !dropdownActive })}
              onChange={(value) => handleNameChange(value.target.value)}
              onClick={showDropdownOption}
              placeholder={t('occurrenceForm.occurrenceData.scientificName')}
              type="text"
              value={name}/>
            {nameValid && !loading
              ? <span className="clear icon is-small is-right"><FontAwesomeIcon className="check" icon="check"/></span>
              : null}
            <span className="invalid-worms"></span>
            <p className="worms-info">Species names are retrieved on the fly  from the WORMS database by using the webservices
              available at <a href="http://www.marinespecies.org/aphia.php?p=webservice" target="_blank">marinespecies.org</a></p>
          </div>
        </div>
        <div className="dropdown-menu" id="dropdown-menu" role="menu">
          <div className="dropdown-content">
            {suggestions.map(record => (
              <div
                className={classNames('dropdown-item', { 'is-active': isRecordWithName(record, name) })}
                key={record.lsid}
                onClick={() => handleSuggestionClick(record.scientificname)}>
                {record.scientificname}
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}

ScientificNameInput.propTypes = {
  onChange:       PropTypes.func.isRequired,
  scientificName: PropTypes.string.isRequired
}
