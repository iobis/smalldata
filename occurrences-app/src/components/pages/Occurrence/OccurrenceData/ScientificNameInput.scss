.scientific-name-input {
  .dropdown-menu {
    margin-top: -1.25rem;
  }
}
