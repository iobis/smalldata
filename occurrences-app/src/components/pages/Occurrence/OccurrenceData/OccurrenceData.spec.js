import React from 'react'
import { mount } from 'enzyme'
import OccurrenceData from './OccurrenceData'

describe('OccurrenceData', () => {
  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })
})

function createComponent() {
  return (
    <OccurrenceData
      data={{
        basisOfRecord:    'humanObservation',
        lifeStage:        'larva',
        occurrenceStatus: 'absent',
        scientificName:   'scientific name',
        scientificNameId: 'scientific:name:id',
        sex:              'male',
        identificationQualifier: 'identification qualifier',
        identificationRemarks: 'identification remarks'
      }}
      onChange={jest.fn()}/>
  )
}
