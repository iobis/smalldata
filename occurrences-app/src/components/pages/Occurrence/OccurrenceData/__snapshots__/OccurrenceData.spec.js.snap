// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OccurrenceData renders correctly 1`] = `
<OccurrenceData
  data={
    Object {
      "basisOfRecord": "humanObservation",
      "lifeStage": "larva",
      "occurrenceStatus": "absent",
      "scientificName": "scientific name",
      "scientificNameId": "scientific:name:id",
      "sex": "male",
    }
  }
  onChange={[MockFunction]}
>
  <div
    className="occurrence-data section is-fluid"
  >
    <div
      className="columns mandatory"
    >
      <ScientificNameInput
        onChange={[Function]}
        onSuggestionClick={[Function]}
        scientificName="scientific name"
      >
        <div
          className="scientific-name-input field column is-four-fifths"
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <label
                className="label"
              >
                occurrenceForm.occurrenceData.scientificName
              </label>
              <div
                className="control has-icons-right is-loading"
              >
                <input
                  className="input scientific-name"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="occurrenceForm.occurrenceData.scientificName"
                  type="text"
                  value="scientific name"
                />
                <span
                  className="invalid-worms"
                />
                <p
                  className="worms-info"
                >
                  occurrenceForm.occurrenceData.scientificNameInput.wormsInfo
                  <a
                    href="http://www.marinespecies.org/aphia.php?p=webservice"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    marinespecies.org
                  </a>
                </p>
              </div>
            </div>
            <div
              className="dropdown-menu"
              id="dropdown-menu"
              role="menu"
            >
              <div
                className="dropdown-content"
              />
            </div>
          </div>
        </div>
      </ScientificNameInput>
    </div>
    <div
      className="columns no-margin"
    >
      <div
        className="column"
      >
        <label
          className="label"
        >
          occurrenceForm.occurrenceData.scientificNameId
        </label>
        <div>
          scientific:name:id
        </div>
      </div>
    </div>
    <InputRadioGroup
      name="occurrenceForm.occurrenceData.occurrenceStatus"
      onChange={[Function]}
      options={
        Array [
          "absent",
          "present",
        ]
      }
      selectedValue="absent"
    >
      <div
        className="input-radio-group field"
      >
        <label
          className="label"
        >
          occurrenceForm.occurrenceData.occurrenceStatus.title
        </label>
        <div
          className="control"
        >
          <InputRadio
            checked={true}
            key="absent"
            name="occurrenceForm.occurrenceData.occurrenceStatus"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.occurrenceStatus.absent"
            value="absent"
          >
            <input
              checked={true}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.occurrenceStatus-absent"
              name="occurrenceForm.occurrenceData.occurrenceStatus"
              onChange={[Function]}
              type="radio"
              value="absent"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.occurrenceStatus-absent"
            >
              occurrenceForm.occurrenceData.occurrenceStatus.absent
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="present"
            name="occurrenceForm.occurrenceData.occurrenceStatus"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.occurrenceStatus.present"
            value="present"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.occurrenceStatus-present"
              name="occurrenceForm.occurrenceData.occurrenceStatus"
              onChange={[Function]}
              type="radio"
              value="present"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.occurrenceStatus-present"
            >
              occurrenceForm.occurrenceData.occurrenceStatus.present
            </label>
          </InputRadio>
        </div>
      </div>
    </InputRadioGroup>
    <InputRadioGroup
      name="occurrenceForm.occurrenceData.basisOfRecord"
      onChange={[Function]}
      options={
        Array [
          "humanObservation",
          "machineObservation",
          "fossilSpecimen",
          "livingSpecimen",
          "preservedSpecimen",
        ]
      }
      selectedValue="humanObservation"
    >
      <div
        className="input-radio-group field"
      >
        <label
          className="label"
        >
          occurrenceForm.occurrenceData.basisOfRecord.title
        </label>
        <div
          className="control"
        >
          <InputRadio
            checked={true}
            key="humanObservation"
            name="occurrenceForm.occurrenceData.basisOfRecord"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.basisOfRecord.humanObservation"
            value="humanObservation"
          >
            <input
              checked={true}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.basisOfRecord-humanObservation"
              name="occurrenceForm.occurrenceData.basisOfRecord"
              onChange={[Function]}
              type="radio"
              value="humanObservation"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.basisOfRecord-humanObservation"
            >
              occurrenceForm.occurrenceData.basisOfRecord.humanObservation
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="machineObservation"
            name="occurrenceForm.occurrenceData.basisOfRecord"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.basisOfRecord.machineObservation"
            value="machineObservation"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.basisOfRecord-machineObservation"
              name="occurrenceForm.occurrenceData.basisOfRecord"
              onChange={[Function]}
              type="radio"
              value="machineObservation"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.basisOfRecord-machineObservation"
            >
              occurrenceForm.occurrenceData.basisOfRecord.machineObservation
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="fossilSpecimen"
            name="occurrenceForm.occurrenceData.basisOfRecord"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.basisOfRecord.fossilSpecimen"
            value="fossilSpecimen"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.basisOfRecord-fossilSpecimen"
              name="occurrenceForm.occurrenceData.basisOfRecord"
              onChange={[Function]}
              type="radio"
              value="fossilSpecimen"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.basisOfRecord-fossilSpecimen"
            >
              occurrenceForm.occurrenceData.basisOfRecord.fossilSpecimen
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="livingSpecimen"
            name="occurrenceForm.occurrenceData.basisOfRecord"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.basisOfRecord.livingSpecimen"
            value="livingSpecimen"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.basisOfRecord-livingSpecimen"
              name="occurrenceForm.occurrenceData.basisOfRecord"
              onChange={[Function]}
              type="radio"
              value="livingSpecimen"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.basisOfRecord-livingSpecimen"
            >
              occurrenceForm.occurrenceData.basisOfRecord.livingSpecimen
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="preservedSpecimen"
            name="occurrenceForm.occurrenceData.basisOfRecord"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.basisOfRecord.preservedSpecimen"
            value="preservedSpecimen"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.basisOfRecord-preservedSpecimen"
              name="occurrenceForm.occurrenceData.basisOfRecord"
              onChange={[Function]}
              type="radio"
              value="preservedSpecimen"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.basisOfRecord-preservedSpecimen"
            >
              occurrenceForm.occurrenceData.basisOfRecord.preservedSpecimen
            </label>
          </InputRadio>
        </div>
      </div>
    </InputRadioGroup>
    <InputRadioGroup
      name="occurrenceForm.occurrenceData.sex"
      onChange={[Function]}
      options={
        Array [
          "male",
          "female",
          "unspecified",
        ]
      }
      selectedValue="male"
    >
      <div
        className="input-radio-group field"
      >
        <label
          className="label"
        >
          occurrenceForm.occurrenceData.sex.title
        </label>
        <div
          className="control"
        >
          <InputRadio
            checked={true}
            key="male"
            name="occurrenceForm.occurrenceData.sex"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.sex.male"
            value="male"
          >
            <input
              checked={true}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.sex-male"
              name="occurrenceForm.occurrenceData.sex"
              onChange={[Function]}
              type="radio"
              value="male"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.sex-male"
            >
              occurrenceForm.occurrenceData.sex.male
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="female"
            name="occurrenceForm.occurrenceData.sex"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.sex.female"
            value="female"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.sex-female"
              name="occurrenceForm.occurrenceData.sex"
              onChange={[Function]}
              type="radio"
              value="female"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.sex-female"
            >
              occurrenceForm.occurrenceData.sex.female
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="unspecified"
            name="occurrenceForm.occurrenceData.sex"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.sex.unspecified"
            value="unspecified"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.sex-unspecified"
              name="occurrenceForm.occurrenceData.sex"
              onChange={[Function]}
              type="radio"
              value="unspecified"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.sex-unspecified"
            >
              occurrenceForm.occurrenceData.sex.unspecified
            </label>
          </InputRadio>
        </div>
      </div>
    </InputRadioGroup>
    <InputRadioGroup
      name="occurrenceForm.occurrenceData.lifeStage"
      onChange={[Function]}
      options={
        Array [
          "larva",
          "juvenile",
          "adult",
          "unspecified",
        ]
      }
      selectedValue="larva"
    >
      <div
        className="input-radio-group field"
      >
        <label
          className="label"
        >
          occurrenceForm.occurrenceData.lifeStage.title
        </label>
        <div
          className="control"
        >
          <InputRadio
            checked={true}
            key="larva"
            name="occurrenceForm.occurrenceData.lifeStage"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.lifeStage.larva"
            value="larva"
          >
            <input
              checked={true}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.lifeStage-larva"
              name="occurrenceForm.occurrenceData.lifeStage"
              onChange={[Function]}
              type="radio"
              value="larva"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.lifeStage-larva"
            >
              occurrenceForm.occurrenceData.lifeStage.larva
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="juvenile"
            name="occurrenceForm.occurrenceData.lifeStage"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.lifeStage.juvenile"
            value="juvenile"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.lifeStage-juvenile"
              name="occurrenceForm.occurrenceData.lifeStage"
              onChange={[Function]}
              type="radio"
              value="juvenile"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.lifeStage-juvenile"
            >
              occurrenceForm.occurrenceData.lifeStage.juvenile
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="adult"
            name="occurrenceForm.occurrenceData.lifeStage"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.lifeStage.adult"
            value="adult"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.lifeStage-adult"
              name="occurrenceForm.occurrenceData.lifeStage"
              onChange={[Function]}
              type="radio"
              value="adult"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.lifeStage-adult"
            >
              occurrenceForm.occurrenceData.lifeStage.adult
            </label>
          </InputRadio>
          <InputRadio
            checked={false}
            key="unspecified"
            name="occurrenceForm.occurrenceData.lifeStage"
            onChange={[Function]}
            text="occurrenceForm.occurrenceData.lifeStage.unspecified"
            value="unspecified"
          >
            <input
              checked={false}
              className="input-radio is-checkradio"
              id="occurrenceForm.occurrenceData.lifeStage-unspecified"
              name="occurrenceForm.occurrenceData.lifeStage"
              onChange={[Function]}
              type="radio"
              value="unspecified"
            />
            <label
              htmlFor="occurrenceForm.occurrenceData.lifeStage-unspecified"
            >
              occurrenceForm.occurrenceData.lifeStage.unspecified
            </label>
          </InputRadio>
        </div>
      </div>
    </InputRadioGroup>
    <div
      className="columns"
    >
      <InputText
        className="identification-qualifier is-9"
        name="occurrenceForm.occurrenceData.identificationQualifier"
        onChange={[Function]}
      >
        <div
          className="column field identification-qualifier is-9"
        >
          <label
            className="label"
            htmlFor="occurrenceform.occurrencedata.identificationqualifier.label"
          >
            occurrenceForm.occurrenceData.identificationQualifier.label
          </label>
          <input
            className="input"
            id="occurrenceform.occurrencedata.identificationqualifier.label"
            onChange={[Function]}
            type="text"
          />
        </div>
      </InputText>
    </div>
    <div
      className="columns"
    >
      <Textarea
        className="identification-remarks is-9"
        name="occurrenceForm.occurrenceData.identificationRemarks"
        onChange={[Function]}
      >
        <div
          className="column field identification-remarks is-9"
        >
          <label
            className="label"
            htmlFor="occurrenceform.occurrencedata.identificationremarks.label"
          >
            occurrenceForm.occurrenceData.identificationRemarks.label
          </label>
          <textarea
            className="textarea"
            id="occurrenceform.occurrencedata.identificationremarks.label"
            onChange={[Function]}
            rows={5}
          />
        </div>
      </Textarea>
    </div>
  </div>
</OccurrenceData>
`;
